bot/exts/evergreen/issues.py

```diff
@@ -17,40 +17,55 @@
     Tokens,
     WHITELISTED_CHANNELS
 )
+from bot.utils.decorators import whitelist_override
+from bot.utils.extensions import invoke_help_command
 
 log = logging.getLogger(__name__)
 
 BAD_RESPONSE = {
     404: "Issue/pull request not located! Please enter a valid number!",
     403: "Rate limit has been hit! Please try again later!"
 }
-
-MAX_REQUESTS = 10
 REQUEST_HEADERS = dict()
 
-REPOS_API = "https://api.github.com/orgs/{org}/repos"
+REPOSITORY_ENDPOINT = "https://api.github.com/orgs/{org}/repos?per_page=100"
+ISSUE_ENDPOINT = "https://api.github.com/repos/{user}/{repository}/issues/{number}"
+PR_MERGE_ENDPOINT = "https://api.github.com/repos/{user}/{repository}/pulls/{number}/merge"
+
 if GITHUB_TOKEN := Tokens.github:
     REQUEST_HEADERS["Authorization"] = f"token {GITHUB_TOKEN}"
 
 WHITELISTED_CATEGORIES = (
     Categories.development, Categories.devprojects, Categories.media, Categories.staff
 )
-WHITELISTED_CHANNELS_ON_MESSAGE = (
-    Channels.organisation, Channels.mod_meta, Channels.mod_tools, Channels.staff_voice
-)
 
 CODE_BLOCK_RE = re.compile(
     r"^`([^`\n]+)`"  # Inline codeblock
     r"|```(.+?)```",  # Multiline codeblock
     re.DOTALL | re.MULTILINE
 )
 
+# Maximum number of issues in one message
+MAXIMUM_ISSUES = 5
+
+
+@dataclass
+class FetchError:
+    """Dataclass representing an error while fetching an issue."""
+
+    return_code: int
+    message: str
+
 
-class FetchIssueErrors(Enum):
-    """Errors returned in fetch issues."""
+@dataclass
+class IssueState:
+    """Dataclass representing the state of an issue."""
 
-    value_error = "Numbers not found."
-    max_requests = "Max requests hit."
+    repository: str
+    number: int
+    url: str
+    title: str
+    icon_url: str
```

Comment:
`icon_url` is conveying the wrong meaning here, it is not a "icon url", but rather an emoji.

Reply:
That was the original name, I don't mind changing it.